apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -576,15 +547,13 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
           tag: "write_on_server_owned_interface"
         )
 
-        ask_clean_session(new_state)
+        {:ok, new_state} = ask_clean_session(new_state, timestamp)
         MessageTracker.discard(new_state.message_tracker, message_id)
```

Comment:
Now that `ask_clean_session` can possibly fail, I'm not sure about the order of these two lines.

The current order gives us Consistency over Availability, while swapping them gives us Availability over Consistency. @bettio what's your opinion?

Reply:
Let's start with Consistency over Availability.